flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1299,6 +1380,39 @@ private static long ceil(long a, long b) {
         }
     }
 
+    private static long ceil(long a, int b) {
+        float q = (float) b / NANOS_PER_MILLISECOND;
```

Comment:
Why do we need it?
Am I right that we want to check just the sign of the result?
At the same time `NANOS_PER_MILLISECOND` is a constant and that means that the sign of the result is the same as the sign of b
so it could be replaces with just something like 
```java
        return b > 0 ? a + 1L : a;
```

Reply:
can you elaborate on this? This is not clear from the code.
If I look at the code for this method
```java
 float q = (float) b / NANOS_PER_MILLISECOND;
if (q > 0) {
    return a + 1L;
} else {
    return a;
}
```
in the first line we are calculating `q` and then use the result in `if` condition
```java
 float q = (float) b / NANOS_PER_MILLISECOND;
if (q > 0) {
 ...
}
```
thus having knowledge about sign of `b` we can answer what is the sign of `q` which is used in `if`.

How can we answer similar question having knowledge about sign of `a` as you suggested?